app/src/test/java/tech/ula/model/repositories/AssetRepositoryTest.kt

```diff
@@ -226,6 +265,58 @@ class AssetRepositoryTest {
         assertTrue(result.isEmpty())
     }
 
+    @Test
+    fun `generateDownloadRequirements does not include rootfs if filesystem needs extraction, there is a local version available, and the network is unreachable`() {
+        val repo = filesystem.distributionType
+        val distDirectory = File("$applicationFilesDirPath/$repo")
+        val assetFile = File("$distDirectory/${distAsset.name}")
+        val rootfsFile = File("$distDirectory/$rootFsTarName")
+        distDirectory.mkdirs()
+        rootfsFile.createNewFile()
+        assetFile.createNewFile()
+
+        whenever(mockAssetPreferences.getLatestDownloadVersion(repo))
+                .thenReturn(lowVersion)
+        runBlocking {
+            whenever(mockGithubApiClient.getLatestReleaseVersion(repo))
+                    .thenThrow(UnknownHostException())
+                    .thenThrow(UnknownHostException())
+        }
+
+        val assetLists = hashMapOf(repo to listOf(distAsset))
+        val filesystemNeedsExtraction = true
+
+        val result = runBlocking {
+            assetRepository.generateDownloadRequirements(filesystem, assetLists, filesystemNeedsExtraction)
+        }
+
+        assertTrue(result.isEmpty())
+    }
+
+    @Test(expected = UnknownHostException::class)
+    fun `generateDownloadRequirements throws UnknownHostException if rootfs files need to be downloaded and the network is unreachable`() {
+        val repo = filesystem.distributionType
+        val distDirectory = File("$applicationFilesDirPath/$repo")
+        val assetFile = File("$distDirectory/${distAsset.name}")
+        distDirectory.mkdirs()
+        assetFile.createNewFile()
+
+        whenever(mockAssetPreferences.getLatestDownloadVersion(repo))
+                .thenReturn(lowVersion)
+        runBlocking {
+            whenever(mockGithubApiClient.getLatestReleaseVersion(repo))
+                    .thenThrow(UnknownHostException())
+                    .thenThrow(UnknownHostException())
```

Comment:
This appears to be an extra line

Reply:
The mock throws once when the stub is during the invocation to `lastDownloadedVersionIsUpToDate(repo)` at line 50, and then again when it is called during the invocation to `lastDownloadedFilesystemVersionIsUpToDate(repo)` at line 71.